scripts/research/b2_05_flow_absorption_lib.py

```
@@ -0,0 +1,1514 @@
+    CANONICAL_SCORE_RECORD_ID order (prereg controls.bootstrap_within_week_
+    sort), not insertion/T order -- the two differ whenever a week mixes
+    BUY and SELL rows, since `side` precedes the timestamp fields in the
+    frozen id. Floating-point addition is not strictly associative, so this
+    ordering is honored exactly rather than assumed immaterial.
+    """
+    values = np.asarray(improvements, dtype=np.float64)
+    stamps = np.asarray(times, dtype=np.int64)
+    if len(values) == 0:
+        return float("nan"), float("nan")
+    groups = defaultdict(list)
+    for value, t, score_id in zip(values, stamps, score_ids):
```

Ruff (0.16.3)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)